src/main/java/uk/gov/gchq/palisade/resource/AbstractLeafResource.java

```diff
@@ -33,13 +33,13 @@
  * See {@link FileResource} for a concrete implementation with an id.
  * This class is mostly used when deserialisation to a LeafResource is required, but the interface can't be used.
  */
-public abstract class AbstractLeafResource extends AbstractResource implements LeafResource, ChildResource {
+public abstract class AbstractLeafResource extends AbstractResource implements LeafResource {
 
     private String type;
     private String serialisedFormat;
     private ConnectionDetail connectionDetail;
     private ParentResource parent;
-    private Map<String, Object> attributes = new HashMap<>();
+    private Map<String, Object> attributes = new LinkedHashMap<>();
```

Comment:
1. HashMap should already be serializable
2. This should still fail SQ because the member var is just a Map

Reply:
Prefer to have a really simple SerializableMap interface extending both Map and Serializable

Reply:
The SerializablePredicate is exactly what I'd expect

Reply:
For some reason it wont serialise if its set to a new HashMap, but it will for LinkedHashMap
@pd104923 wrote a serializableMap that extended map and implemented serializable but we were told to remove it

Reply:
Java very much does not support this.... changed Maps to HashMaps instead